failing test diff is flipped since version 0.32.1

### Describe the bug

the following test: 
```ts
import { describe, test, expect } from 'vitest';

describe('rejects', () => {
  test('buyApples throws an error when no id provideds', async () => {
    // toThrow returns a promise now, so you HAVE to await it
    await expect('abc').toBe('abcd');
  });
});
```
has different failing diff/results for version `0.32.0` and `0.32.1`. `0.32.0` is the correct expected results (at least to me)


`0.32.0` results (_correct_)

```
AssertionError: expected 'abc' to be 'abcd' // Object.is equality

  - Expected  - 0
  + Received  + 1

  - 'abcd'
  + 'abc'
```

`0.32.1` results (_incorrect_)
```
AssertionError: expected 'abc' to be 'abcd' // Object.is equality

- Expected
+ Received

- abc
+ abcd
```

### Reproduction

[Minimal reproduction](https://stackblitz.com/edit/vitest-dev-vitest-1rx27b?file=package.json,suite.test.ts)

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.4.2 - /usr/local/bin/npm
  npmPackages:
    @vitest/ui: 0.31.4 => 0.31.4 
    vite: 4.3.9 => 4.3.9 
    vitest: 0.32.1 => 0.32.1
```


### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/utils/src/diff/index.ts
+++ b/packages/utils/src/diff/index.ts
@@ -60,6 +60,12 @@ const FALLBACK_FORMAT_OPTIONS = {
 // Generate a string that will highlight the difference between two values
 // with green and red. (similar to how github does code diffing)
 
+/**
+ * @param a Expected value
+ * @param b Received value
+ * @param options Diff options
+ * @returns
+ */
 export function diff(a: any, b: any, options?: DiffOptions): string | null {
   if (Object.is(a, b))
     return ''

--- a/packages/utils/src/error.ts
+++ b/packages/utils/src/error.ts
@@ -97,7 +97,7 @@ export function processError(err: any) {
     err.nameStr = String(err.name)
 
   if (err.showDiff || (err.showDiff === undefined && err.expected !== undefined && err.actual !== undefined))
-    err.diff = diff(err.actual, err.expected)
+    err.diff = diff(err.expected, err.actual)
 
   if (typeof err.expected !== 'string')
     err.expected = stringify(err.expected, 10)

--- a/test/core/test/jest-expect.test.ts
+++ b/test/core/test/jest-expect.test.ts
@@ -1,7 +1,9 @@
 /* eslint-disable no-sparse-arrays */
 import { AssertionError } from 'node:assert'
 import { describe, expect, it, vi } from 'vitest'
-import { generateToBeMessage } from '@vitest/expect'
+import { generateToBeMessage, setupColors } from '@vitest/expect'
+import { processError } from '@vitest/utils/error'
+import { getDefaultColors } from '@vitest/utils'
 
 class TestError extends Error {}
 
@@ -751,6 +753,19 @@ it('compatible with jest', () => {
   expect(matchers).toHaveProperty('someObject')
   expect(matchers).toHaveProperty('toBe')
   expect(state.assertionCalls).toBe(2)
+})
+
+it('correctly prints diff', () => {
+  try {
+    expect({ a: 1 }).toEqual({ a: 2 })
+    expect.unreachable()
+  }
+  catch (err) {
+    setupColors(getDefaultColors())
+    const error = processError(err)
+    expect(error.diff).toContain('-   "a": 2')
+    expect(error.diff).toContain('+   "a": 1')
+  }
 })
 
 it('timeout', () => new Promise(resolve => setTimeout(resolve, 500)))

